packages/appium/lib/config.js

```diff
@@ -209,6 +217,46 @@ function getNonDefaultServerArgs (args) {
   return _.pickBy(args, (__, key) => isNotDefault(key));
 }
 
+/**
+ * Compacts an object for {@link showConfig}:
+ * 1. Removes `subcommand` key/value
+ * 2. Removes `undefined` values
+ * 3. Removes empty objects (but not `false` values)
+ * Does not operate recursively.
+ */
+const compactConfig = _.partial(
+  _.omitBy,
+  _,
+  (value, key) => key === 'subcommand' || _.isUndefined(value) || (_.isObject(value) && _.isEmpty(value))
```

Comment:
I assume `_.isObject(value) && _.isEmpty(value)` should filter out undefined values as well

Reply:
no, because `undefined` is not an object